File: app/src/main/java/eu/stuifzand/micropub/auth/WebSigninActivity.java
package eu.stuifzand.micropub.auth;

import android.accounts.AccountAuthenticatorResponse;
import android.accounts.AccountManager;
import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

import eu.stuifzand.micropub.R;


public class WebSigninActivity extends AppCompatActivity {

    public static final int AUTHENTICATION_REQUEST = 14;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_web_signin);
    }


    public void startWebsignin(View view) {
        EditText profileUrl = (EditText) findViewById(R.id.profileUrl);
        String url = profileUrl.getText().toString();

        Intent intent = getIntent();
        AccountAuthenticatorResponse parcelable = intent.getParcelableExtra(AccountManager.KEY_ACCOUNT_AUTHENTICATOR_RESPONSE);
        new WebsigninTask(this, parcelable).execute(url);
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        if (requestCode == AUTHENTICATION_REQUEST) {
            if (resultCode == RESULT_OK) {
                finish();
            }
        }
    }
}

File: app/src/main/java/eu/stuifzand/micropub/auth/WebSigninActivity.kt
package eu.stuifzand.micropub.auth

import android.accounts.AccountAuthenticatorResponse
import android.accounts.AccountManager
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import eu.stuifzand.micropub.R
import kotlinx.android.synthetic.main.activity_web_signin.*

const val AUTHENTICATION_REQUEST = 14;

class WebSigninActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_web_signin)
    }

    fun startWebsignin(view : View) {
        val url = profileUrl.text.toString();
        val me = if (!url.matches(Regex.fromLiteral("^https?://"))) "https://$url" else url
        val parcelable = intent.getParcelableExtra<AccountAuthenticatorResponse>(AccountManager.KEY_ACCOUNT_AUTHENTICATOR_RESPONSE);
        WebsigninTask(this, parcelable).execute(me);
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (requestCode == AUTHENTICATION_REQUEST) {
            if (resultCode == RESULT_OK) {
                finish();
            }
        }
    }
}
